backend/scripts/scraping.py

```diff
@@ -0,0 +1,319 @@
+    """Scrap classes schedule
+
+    Args:
+        path (str, optional): classes schedule as an html. Defaults to 'edt.html'.
+    """
+    with open(path, 'r', encoding='utf-8') as file:
+        html_content = file.read()
+    
+    soup = BeautifulSoup(html_content, 'html.parser') # Parser le contenu HTML avec BeautifulSoup
+
+    # Find the elements containing the relevant information
+    dates = soup.find_all('span', class_='fc-list-heading-main')
+    events = soup.find_all("tr", class_="fc-list-item")
+
+    # Create a dictionary to store events for each date
+    date_events = {}
+    current_date = None  # Initialize the current date to None
+
+    # Iterate through dates and extract events
+    for i in range(len(events)):
+        if dates and events[i].find_previous('span', class_='fc-list-heading-main') == dates[0]:
+            current_date = dates.pop(0).text  # Get the text of the date
+
+        if current_date is not None:
+            event_name = events[i]  # Get the html of the event
+            date_events.setdefault(current_date, []).append(event_name)
+
+    # Initialize a list to store event data
+    event_data = []
+
+    for date, event_list in date_events.items():
+        for event in event_list: # Iterate through the events and extract the information
+            time = event.find("td", class_="fc-list-item-time").get_text(strip=True)
+            title = event.find("td", class_="fc-list-item-title").find("a")
+
+            lesson_info = re.split(r',\s*', title.get_text()) # Split the class title using a comma
+            name = lesson_info[0].split(" (", 1)[0] #Extract class info
+
+            #Extract lesson info
+            time_parts = time.split(" - ") # Extract day and start/end times from the time string
+            start_time = time_parts[0]
+            end_time = time_parts[1]
+
+            lesson_type = lesson_info[0].split()[-1]
+            if not(lesson_type.startswith('Cours')) and not(lesson_type.startswith('TP')) and not(lesson_type.startswith('TD')):
+                lesson_type = "TBD"
+            if lesson_info[-2].isupper():
+                teacher = lesson_info[-2]
+            else:
+                teacher = "TBD"
+
+            room = lesson_info[-1]
+            date_iso = transform_date_french_to_iso(date)
+            event_info = (date_iso, transform_time(start_time), transform_time(end_time), lesson_type, teacher, room)
+
+            background_color = event.find('span', class_='fc-event-dot')['style'].split(':')[1]
+
+            # Create an event dictionary
+            event_dict = {
+                "class_id": "TBD",
+                "name": name,
+                "ects": "TBD",
+                "lesson_info": [event_info],
+                "backgroundColor": background_color,
+                # "link":"link",
+                "teacher_name":"TBD"
+            }
+
+            event_data.append(event_dict)
+
+    return(event_data)
+
+event_data = scrap_events_data()
```

Comment:
No function should be called outside of the if __name__=="__main__" block
either create a main() function in the code here that you call in the __main__ block, or move these in the __main__ block